File: proposed-projects/round-trip-experiment/reverse/20260909T130706Z_31ac2789/20260910T073404Z_10c90cd6/ported_implementation/gbnf/rules_builder/parse_space.py
def parse_space(src: str, pos: int, newline_ok: bool) -> int:
    while pos < len(src) and (
        src[pos] == " "
        or src[pos] == "\t"
        or src[pos] == "#"
        or (newline_ok and (src[pos] == "\r" or src[pos] == "\n"))
    ):
        if src[pos] == "#":
            while pos < len(src) and src[pos] != "\r" and src[pos] != "\n":
                pos += 1
        else:
            pos += 1
    return pos


parseSpace = parse_space

File: packages/gbnf-experiment/data/20260909T182714Z_2643e4be/ported_implementation/src/rules-builder/parse-space.ts
export const parseSpace = (
  src: string,
  pos: number,
  newlineOk: boolean,
): number => {
  while (
    pos < src.length &&
    (src[pos] === " " ||
      src[pos] === "\t" ||
      src[pos] === "#" ||
      (newlineOk && (src[pos] === "\r" || src[pos] === "\n")))
  ) {
    if (src[pos] === "#") {
      while (pos < src.length && src[pos] !== "\r" && src[pos] !== "\n") {
        pos += 1;
      }
    } else {
      pos += 1;
    }
  }
  return pos;
};
